### PR TITLE
update sample repo for 1.0

### PR DESCRIPTION
**Pull request by lilly-tecton**
only remaining issue is materialization context causing destructive upgrade during sdk update: https://linear.app/tecton/issue/FE-2106/materialization-context-causing-destructive-upgrade-during-sdk-update 

```
        pipeline.root.transformation_node.inputs[1].node.context_node.context_type: CONTEXT_TYPE_UNSPECIFIED -> CONTEXT_TYPE_MATERIALIZATION
        pipeline.root.transformation_node.inputs[1].node.context_node.input_name:  -> context
``` 

https://linear.app/tecton/issue/FE-2120/update-the-samples-repo-to-use-the-features-parameter

### CODE REVIEW

File: ads/features/stream_features/content_keyword_clicks_push.py [hunk 1]
```diff
@@ -19,5 +20,10 @@
     ttl=timedelta(days=30),
     tags={'release': 'production'},
     owner='demo-user@tecton.ai',
-    description='The ad clicks for a content keyword'
+    description='The ad clicks for a content keyword',
+    batch_trigger=BatchTriggerType.MANUAL,
```
Comment (lilly-tecton) [inline ads/features/stream_features/content_keyword_clicks_push.py:24]: this had to be added in order for feature view to build, not relevant to features migration

File: ads/features/feature_tables/ad_embeddings.py [hunk 2]
```diff
@@ -1,19 +1,16 @@
-from tecton.types import Field, String, Timestamp, Array, Float64
-from tecton import Entity, FeatureTable, DeltaConfig
-from ads.entities import ad
+from tecton.types import Array, Float64
+from tecton import FeatureTable, Attribute
 from datetime import timedelta
 
-schema = [
-    Field('ad_id', String),
```
Comment (lilly-tecton) [inline ads/features/feature_tables/ad_embeddings.py:7]: i had to apply with ad_id as Int64 in order for this to be consistent with the `ad` entity (which has Int64 in the other feature views)

File: repo.yaml
Comment (lilly-tecton) [inline repo.yaml:1]: i think the repo.yaml requirement is new?

File: requirements.txt [hunk 3]
```diff
@@ -1 +1 @@
-tecton[pyspark]~=0.7.0
+tecton[pyspark]~=0.10.0b30
```
Comment (lilly-tecton) [inline requirements.txt:1]: will update to 1.0 prior to merge

File: fraud/features/batch_features/tests/test_user_credit_card_issuer.py [hunk 4]
```diff
@@ -12,12 +13,22 @@
 # session is needed, then you can create your own and set it with `tecton.set_tecton_spark_session()`.
 @pytest.mark.skipif(os.environ.get("TECTON_TEST_SPARK") is None, reason="Requires JDK installation and $JAVA_HOME env variable to run, so we skip unless user sets the `TECTON_TEST_SPARK` env var.")
 def test_user_credit_card_issuer_run(tecton_pytest_spark_session):
-    input_pandas_df = pandas.DataFrame({
-        "user_id": ["user_1", "user_2", "user_3", "user_4"],
-        "signup_timestamp": [datetime(2022, 5, 1)] * 4,
-        "cc_num": [1000000000000000, 4000000000000000, 5000000000000000, 6000000000000000],
-    })
-    input_spark_df = tecton_pytest_spark_session.createDataFrame(input_pandas_df)
```
Comment (lilly-tecton) [inline fraud/features/batch_features/tests/test_user_credit_card_issuer.py:20]: as a part of the update to 1.0, the version of pandas also updated to > 2.0. the updated version of pandas cannot create dataframes this way (https://stackoverflow.com/questions/75926636/databricks-issue-while-creating-spark-data-frame-from-pandas), so i had to update the way the resulting df was created

File: fraud/features/batch_features/tests/test_user_distinct_merchant_transaction_count_30d.py [hunk 5]
```diff
@@ -34,9 +42,10 @@ def test_user_distinct_merchant_transaction_count_30d(tecton_pytest_spark_sessio
 
     expected = pandas.DataFrame({
         "user_id": ["user_1", "user_2"],
-        "timestamp": [datetime(2022, 5, 2) - timedelta(microseconds=1), datetime(2022, 5, 2) - timedelta(microseconds=1)],
+        "timestamp": [pytz.UTC.localize(datetime(2022, 5, 2) - timedelta(microseconds=1)), pytz.UTC.localize(datetime(2022, 5, 2) - timedelta(microseconds=1))],
         "distinct_merchant_transaction_count_30d": [2, 1],
     })
+    expected['timestamp'] = expected['timestamp'].astype('datetime64[us, UTC]')
```
Comment (lilly-tecton) [inline fraud/features/batch_features/tests/test_user_distinct_merchant_transaction_count_30d.py:48]: maybe there's a more graceful way to do this

File: ads/features/stream_features/content_keyword_click_counts.py [hunk 6]
```diff
@@ -10,22 +13,24 @@
...
     batch_compute=cluster_config,
     stream_compute=cluster_config,
     online=False,
     offline=False,
     feature_start_time=datetime(2022, 5, 1),
     tags={'release': 'production'},
     owner='demo-user@tecton.ai',
-    description='The count of ad impressions for a content_keyword'
+    description='The count of ad impressions for a content_keyword',
+    aggregation_leading_edge=AggregationLeadingEdge.LATEST_EVENT_TIME
```
Comment (njoung) [inline ads/features/stream_features/content_keyword_click_counts.py:33]: Wondering if we should go back and update this eventually to the newest param

File: fraud/features/batch_features/tests/test_user_credit_card_issuer.py [hunk 7]
```diff
@@ -29,22 +40,31 @@ def test_user_credit_card_issuer_run(tecton_pytest_spark_session):
 
     expected = pandas.DataFrame({
         "user_id": ["user_1", "user_2", "user_3", "user_4"],
-        "signup_timestamp":  [datetime(2022, 5, 1)] * 4,
+        "signup_timestamp":  [pytz.UTC.localize(datetime(2022, 5, 1, 0, 0, 0, 0))] * 4,
```
Comment (njoung) [inline fraud/features/batch_features/tests/test_user_credit_card_issuer.py:43]: Why was this necessary? I think `datetime(2022, 5, 1 tzinfo='utc')` should be enough?

File: fraud/features/batch_features/tests/test_user_credit_card_issuer.py [hunk 8]
```diff
@@ -29,22 +40,31 @@ def test_user_credit_card_issuer_run(tecton_pytest_spark_session):
...
+        ("user_1", signup_timestamp, 1000000000000000),
+        ("user_2", signup_timestamp, 4000000000000000),
+        ("user_3", signup_timestamp, 5000000000000000),
+        ("user_4", signup_timestamp, 6000000000000000)
+    ]
+
+    # Define schema: user_id, signup_timestamp, cc_num
+    schema = ["user_id", "signup_timestamp", "cc_num"]
+
+    # Convert to Spark DataFrame
```
Comment (njoung) [inline fraud/features/batch_features/tests/test_user_credit_card_issuer.py:66]:
```suggestion
    schema = ["user_id", "signup_timestamp", "cc_num"]

```

File: fraud/features/batch_features/tests/test_user_distinct_merchant_transaction_count_30d.py (hunk 5 above)
Comment (njoung) [reply fraud/features/batch_features/tests/test_user_distinct_merchant_transaction_count_30d.py:48]: You should be able to attach the astype around the pandas.DataFrame creation but otherwise this is just what I've been doing

File: fraud/features/batch_features/tests/test_user_credit_card_issuer.py (hunk 7 above)
Comment (lilly-tecton) [reply fraud/features/batch_features/tests/test_user_credit_card_issuer.py:43]: maybe. i just asked chatgpt how to fix this

Comment (lilly-tecton) [reply fraud/features/batch_features/tests/test_user_credit_card_issuer.py]: fixed

File: repo.yaml [hunk 9]
```diff
@@ -11,13 +11,13 @@
 
 defaults:
   batch_feature_view:
-    tecton_materialization_runtime: 0.9.0
-    environment: tecton-rift-core-0.9.0  # For Rift-based Batch Feature Views
+    tecton_materialization_runtime: 1.0.0
+    environment: tecton-rift-core-1.0.0  # For Rift-based Batch Feature Views
```
Comment (njoung) [inline repo.yaml:15]: out of curiosity do we need to set this here?

File: fraud/features/stream_features/tests/test_user_recent_transactions.py [hunk 10]
```diff
@@ -15,15 +16,17 @@
...
-    })
-    input_spark_df = my_custom_spark_session.createDataFrame(input_pandas_df)
+    data = [
+        ("user_1", datetime(2022, 5, 1), 100, "2022", "05", "01"),
+        ("user_1", datetime(2022, 5, 1), 200, "2022", "05", "01"),
+        ("user_1", datetime(2022, 5, 1), 300, "2022", "05", "01"),
+        ("user_2", datetime(2022, 5, 1), 400, "2022", "05", "01")
+    ]
+
+    # Define the schema (column names)
```
Comment (njoung) [inline fraud/features/stream_features/tests/test_user_recent_transactions.py:26]:
```suggestion
```

File: ads/features/stream_features/content_keyword_click_counts.py (hunk 6 above)
Comment (njoung) [reply ads/features/stream_features/content_keyword_click_counts.py:33]: Yeah just a musing, I know there's a preference for wallclock time but out of scope for now https://docs.tecton.ai/changelog/1.0.0#aggregation-leading-edge-parameter

File: repo.yaml (hunk 9 above)
Comment (lilly-tecton) [reply repo.yaml:15]: not sure. can you try removing?

Comment (lilly-tecton) [reply repo.yaml:15]: removed

**Review by njoung — APPROVED** [review]
